$OpenBSD: patch-src_sqtop_cpp,v 1.2 2018/04/08 19:38:07 jasper Exp $

- error: cannot initialize a member subobject of type 'int' with an rvalue of type 'nullptr_t'
- Revision: 21859c656dbc: sqtop: set resolving opts before getting info from squid

Index: src/sqtop.cpp
--- src/sqtop.cpp.orig
+++ src/sqtop.cpp
@@ -35,7 +35,7 @@ static struct option longopts[] = {
    { "hosts",              required_argument,   NULL,    'H' },
    { "users",              required_argument,   NULL,    'u' },
    { "pass",               required_argument,   NULL,    'P' },
-   { "help",               no_argument,         NULL,    NULL},
+   { "help",               no_argument,         NULL,    0   },
    { "brief",              no_argument,         NULL,    'b' },
    { "full",               no_argument,         NULL,    'f' },
    { "zero",               no_argument,         NULL,    'z' },
@@ -266,6 +266,10 @@ int main(int argc, char **argv) {
 #endif
       sqstat sqs;
       std::vector<SQUID_Connection> stat;
+#ifdef WITH_RESOLVER
+      pOpts->pResolver->resolve_mode = Resolver::RESOLVE_SYNC;
+#endif
+      pOpts->speed_mode = Options::SPEED_AVERAGE;
       try {
          stat = sqs.GetInfo(pOpts);
       }
@@ -273,11 +277,7 @@ int main(int argc, char **argv) {
          cerr << e.what() << endl;
          exit(1);
       }
-      pOpts->speed_mode = Options::SPEED_AVERAGE;
       cout << sqstat::HeadFormat(pOpts, sqs.active_conn, stat.size(), sqs.av_speed) << endl;
-#ifdef WITH_RESOLVER
-      pOpts->pResolver->resolve_mode = Resolver::RESOLVE_SYNC;
-#endif
       cout << conns_format(pOpts, stat) << endl;
 #ifdef ENABLE_UI
    }
